File: src/main/java/com/wire/xenon/user/model/Access.java
package com.wire.xenon.user.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Access {
    private Cookie cookie;

    @JsonProperty("user")
    public UUID userId;

    @JsonProperty("access_token")
    public String token;

    @JsonProperty("expires_in")
    public int expire;

    @JsonProperty("token_type")
    public String type;

    public UUID getUserId() {
        return userId;
    }

    public String getToken() {
        return token;
    }

    public int getExpire() {
        return expire;
    }

    public String getType() {
        return type;
    }

    public Cookie getCookie() {
        return cookie;
    }

    public void setCookie(Cookie cookie) {
        this.cookie = cookie;
    }
}

File: src/main/java/com/wire/kalium/helium/models/Access.kt
package com.wire.kalium.helium.models

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
class Access {
    @set:JsonIgnore
    @JsonIgnore
    var cookie: Cookie? = null

    @JsonProperty("user")
    var userId: UUID? = null

    @JsonProperty("access_token")
    var accessToken: String? = null

    @JsonProperty("expires_in")
    var expiresIn = 0

    @JsonProperty("token_type")
    var tokenType: String? = null

    @JsonIgnore
    fun hasCookie(): Boolean {
        return cookie != null
    }
}
